opm/material/thermal/EclThermalLawManager.hpp

```diff
@@ -69,29 +69,33 @@ class EclThermalLawManager
         thermalConductivityApproach_ = ThermalConductionLawParams::undefinedApproach;
     }
 
-    void initFromDeck(const Opm::Deck& deck,
-                      const Opm::EclipseState& eclState,
-                      const std::vector<int>& compressedToCartesianElemIdx)
+    void initFromDeck(const Opm::Deck& deck)
+    {
+        hasSpecRock = deck.hasKeyword("SPECROCK");
```

Comment:
I **think** it's possible to answer the same question from `EclipseState` too.  Something along the lines of
```C++
hasSpecRock = eclState.getTableManager().hasTables("SPECROCK");
```
We'd need to check that, obviously, but it might be something to keep in mind if we really want to reduce the occurrence of direct access to the `Deck` class.

Reply:
ultimately i think the simulators should not have access to the deck, ie, we can discard it after setting up the eclipsestate, schedule and field props. i'll try this one for sure.

Reply:
does the job so this PR and downstream is slightly altered now.